New icon detection in general + support for additional mimetype icons

AndyScherzinger opened the pull request:
for details and discussion, please see  #1118.

example screenshot showing some of the additional icons now incorporated:
![device-2015-09-10-180148](https://cloud.githubusercontent.com/assets/1315170/9793605/7cd5d6f2-57e6-11e5-8165-8a6c3fe3eb1e.png)


<!---
@huboard:{"order":1.28125,"milestone_order":1149,"custom_state":""}
-->

AndyScherzinger commented:
@rperezb this PR implements a rewritten icon determination based on the web app for 8.2, so this is for further visual consistency.

@rullzer the lists might be something worth adapting in the web app, since it has some more mappings for the mime types to show the proper icon, e.g. for the iWorks suite. :)

davivel commented on src/com/owncloud/android/utils/MimetypeIconUtil.java:
```diff
@@ -0,0 +1,411 @@
+        MIMETYPE_TO_ICON_MAPPING.put("application/vnd.oasis.opendocument.text-master", R.drawable.file_doc);
+        MIMETYPE_TO_ICON_MAPPING.put("application/vnd.oasis.opendocument.text-template", R.drawable.file_doc);
+        MIMETYPE_TO_ICON_MAPPING.put("application/vnd.oasis.opendocument.text-web", R.drawable.file_doc);
+        MIMETYPE_TO_ICON_MAPPING.put("application/vnd.openxmlformats-officedocument.presentationml.presentation", R.drawable.file_ppt);
```
Lines should be kept under 120 characters to avoid the need of scrolling horizontally in Github

AndyScherzinger replied:
I know, should we also do this for lines that put objects into a map? If so I will reformat :)

davivel commented on src/com/owncloud/android/utils/MimetypeIconUtil.java:
```diff
@@ -0,0 +1,411 @@
+
+    /**
+     public static void main(String[] args) {
+     Iterator<String> fileExtensions = FILE_EXTENSION_TO_MIMETYPE_MAPPING.keySet().iterator();
```
Please, remove commented code.

AndyScherzinger replied:
DONE!

davivel commented:
Really nice work.

:+1:

By the way, please, disable automatic formatting in Android Studio. Not in this case, but some times it makes reviews really hard.

AndyScherzinger commented:
@davivel should I reformat the lines for 120 for the hasmap population, or leave it like that?

(I know it makes reviews on github slightly harder, bur the code (for this part) seems to be more readable)

davivel commented:
Yes, please. It's not more readable in Github.

Just the guilty lines, no need to reformat all the map.

AndyScherzinger commented:
hmm, all the other lines actually are below 120 or I am missing something.

davivel commented:
That's fine, all the other lines are below 120.

I just wanted to remark that you don't need to change the indentation of all the lines in the hashmap assignment to have all of them in a single format. Sorry for the confusion.

AndyScherzinger commented:
Ah, okay, got it :)
Thanks for the review (at this time of the day!).

davivel commented:
You're welcome, but we have more to thank you than on the other way :)

Tomorrow, more.

masensio commented:
Hi @AndyScherzinger,

I'm checking this PR. I have a folder with xml files, but the icon is different in the server. You can see in the next image:

Server filelist:
![filelist-server01](https://cloud.githubusercontent.com/assets/3862330/9877585/10ebbe26-5bbe-11e5-8e32-17ad698aeaf3.png)

Device Filelist:
![screenshot_2015-09-15-15-17-39](https://cloud.githubusercontent.com/assets/3862330/9877613/39735106-5bbe-11e5-9a82-b99af9e62010.png)

Could you review the implementation for xml files?

AndyScherzinger commented:
That is true, in contrast to the web implementation I chose application/xml als the mimetype (which is the true one) and used a different mapping, since I considered xml to be code rather than text.

I  can change that if you want me to :)

davivel commented:
Yes, please. Let's try to keep it consistent with the server for this time.

AndyScherzinger commented:
@davivel and @masensio how should it be changed? What I can do is also fall back to the doc icon, but right now @Kernald added this in blue, so it will be the same symbol, but a different color. Did they change this very recently for the web version?!

Kernald commented:
The `code` icon used in the Android screenshot is the one currently named [text-code](https://github.com/owncloud/core/blob/master/core/img/filetypes/text-code.png) on the server.

AndyScherzinger commented:
@Kernald point is they want us to use the https://github.com/owncloud/core/blob/master/core/img/filetypes/text.png icon for that (which is grey) on the server now. (while Android is blue at the moment)

AndyScherzinger commented:
@Kernald can you open a PR with the text icon in its various sizes against the `new_filetype_icons`branch? I'll merge it and change the detection code :)

Kernald commented:
I'll try to do it in the next couple of days ;-)

davivel commented:
Sorry, in this case we can't wait so much.

I will do it.

AndyScherzinger commented:
Awesome @davivel ! :)

AndyScherzinger commented:
I can do the code change, if you give me the name of the png file you will add :)

AndyScherzinger commented:
@davivel DONE, I changed the code and added a copy of file_doc --> file_text. Just replace file_text in the xxhdpi folder and add the appropriate ones in the other resolutions folders and you should be done :)

(I added a dummy for the code to compile)

davivel commented:
There it is.

Thanks!

AndyScherzinger commented:
works at least for txt and xml:
![device-2015-09-15-181433](https://cloud.githubusercontent.com/assets/1315170/9882623/ea44c548-5bd5-11e5-8dbc-1127c8fcdb12.png)

AndyScherzinger commented:
Will this make it into 1.8? Thought we already have a freeze. (would love to have it in 1.8.0 anyways :grin: )

davivel commented:
That's still the idea. We are rebasing the release branch for these last changes. But this has to come to and end ASAP. 

Hopefully this will be the last change to take into account. The milestone is now updated, nothing else should go in.

AndyScherzinger commented:
I didn't want to push this in and I would have been fine with leaving this out. I just went on implementing further stuff (like this :+1: )

davivel commented:
Hey, I didn't mean this was due to your comments. No blaming, pal, on the contrary, great work here on your side. +1000 :-)

Just want to let it clear in case anybody else reads the thread and wants to push their own stuff. The decision was not due to your input, but due to the topic itself and the previous merge of the fist update of icons.

rullzer commented:
> @rullzer the lists might be something worth adapting in the web app, since it has some more mappings for the mime types to show the proper icon, e.g. for the iWorks suite. :)

@AndyScherzinger I leave that up to @jancborchardt :)

AndyScherzinger commented:
Sure thing! Maybe in a post 8.2 release and Android App 1.8.1 or so. I agree with the Android Team that 1.8.0 is frozen

jancborchardt commented:
@rullzer @AndyScherzinger I’d have to ask one of you to do the mapping since I have very little technical understanding of how to add it. ;) I only make the icons.

AndyScherzinger commented:
I think @rullzer had the point that adding new Mappings would be your decision. We can do the adding of course :)

jancborchardt commented:
Aah no. I trust you know what you are doing of course. :D And adding iWorks filetype mapping sounds like it makes a lot of sense.

masensio commented:
Hi @AndyScherzinger,

I've reviewed again this PR, with the most of kind of files (more than 100 extensions) and the icons are consistent with the kind of file.  So I'll merge this PR.

New nice icons for 1.8 version!! :smiley: 

Cool work!

AndyScherzinger commented:
awesome! :+1:

AndyScherzinger commented:
Wouldn't this then also have to be pushed to https://github.com/owncloud/android/tree/release-1.8 ?

davivel commented:
Done; thanks, @AndyScherzinger
